lms/djangoapps/course_api/blocks/utils.py

```diff
@@ -0,0 +1,22 @@
+"""
+   Utils for Blocks
+"""
+from openedx.core.djangoapps.discussions.models import (
+    DiscussionsConfiguration,
+    Provider,
+)
+
+
+def filter_discussion_xblocks_from_response(response, course_key, request_from_mobile=False):
+    """
+    Removes discussion xblocks if request is from mobile and provider is openedx
+    """
+    configuration = DiscussionsConfiguration.get(context_key=course_key)
+    provider = configuration.provider_type
+    if provider == Provider.OPEN_EDX and request_from_mobile:
```

Comment:
I think we do not need this request_from_mobile condition. If the provider is OPEN_EDX then we do not want to return the discussion xblocks in any case because they are no longer relevant